### ISSUE TITLE
New labelers are not executable

### ISSUE THREAD

**Issue by manuelgitgomes** [lardemua/atom#805]
When running `collect_data.launch`, I encountered this error:

![image](https://github.com/lardemua/atom/assets/58152228/85ef4d89-6728-4772-9eec-217742453b0c)

Should be an easy fix, just giving these files the permission to execute.

**Comment by miguelriemoliveira** [comment]
Hi @manuelgitgomes,

thanks. Please, whenever possible, copy paste text instead of an image. It is always better to have text.

**Comment by miguelriemoliveira** [comment]
About the issue:

They are executable from my side, so I am not really sure what to do.

```terminal
➜  collect git:(noetic-devel) ✗ pwd
/home/mike/workspaces/catkin_ws/src/calibration/atom/atom_calibration/src/atom_calibration/collect
➜  collect git:(noetic-devel) ✗ ls -lah
total 152K
drwxrwxr-x 3 mike mike 4.0K Dec 23 15:01 .
drwxrwxr-x 7 mike mike 4.0K Mar 29  2023 ..
-rw-rw-r-- 1 mike mike  982 Nov 20 23:26 configurable_tf_listener.py
-rw-r--r-- 1 mike mike  31K Dec 24 12:29 data_collector.py
-rwxrwxrwx 1 mike mike 8.8K Dec 22 20:30 depth_labeler
-rw-rw-r-- 1 mike mike    0 Mar 28  2023 __init__.py
-rw-r--r-- 1 mike mike  44K Dec 22 20:30 label_messages.py
-rwxrwxrwx 1 mike mike  12K Dec 22 20:30 lidar3d_labeler
-rw-r--r-- 1 mike mike  23K Dec 23 09:24 patterns.py
drwxrwxr-x 2 mike mike 4.0K Dec 24 12:29 __pycache__
-rwxrwxrwx 1 mike mike 8.5K Dec 22 20:30 rgb_labeler
➜  collect git:(noetic-devel) ✗ 
```

Can you try to make them executable and see if that solves it. 

I don't think this will sove it for everyone tough ...

**Comment by manuelgitgomes** [comment]
The files on my side were:


```
-rw-r--r-- 1 nel nel  982 Nov 23 10:28 configurable_tf_listener.py
-rw-r--r-- 1 nel nel  31K Dec 28 16:32 data_collector.py
-rw-r--r-- 1 nel nel 8.8K Dec 28 16:32 depth_labeler
-rw-rw-r-- 1 nel nel    0 Jan 26  2023 __init__.py
-rw-r--r-- 1 nel nel  44K Dec 28 16:32 label_messages.py
-rw-r--r-- 1 nel nel  12K Dec 28 16:32 lidar3d_labeler
-rw-r--r-- 1 nel nel  23K Dec 28 16:32 patterns.py
drwxrwxr-x 2 nel nel 4.0K Jan  5 09:54 __pycache__
-rw-r--r-- 1 nel nel 8.5K Dec 28 16:32 rgb_labeler

```

After giving them correct permissions:

```
-rw-r--r-- 1 nel nel  982 Nov 23 10:28 configurable_tf_listener.py
-rw-r--r-- 1 nel nel  31K Dec 28 16:32 data_collector.py
-rwxr-xr-x 1 nel nel 8.8K Dec 28 16:32 depth_labeler
-rw-rw-r-- 1 nel nel    0 Jan 26  2023 __init__.py
-rw-r--r-- 1 nel nel  44K Dec 28 16:32 label_messages.py
-rwxr-xr-x 1 nel nel  12K Dec 28 16:32 lidar3d_labeler
-rw-r--r-- 1 nel nel  23K Dec 28 16:32 patterns.py
drwxrwxr-x 2 nel nel 4.0K Jan  5 09:54 __pycache__
-rwxr-xr-x 1 nel nel 8.5K Dec 28 16:32 rgb_labeler

```

Everything ran smoothly after this.
I can investigate and see if this is a common issue and how to solve it.

**Comment by miguelriemoliveira** [comment]
ok

**Comment by manuelgitgomes** [comment]
> They are executable from my side, so I am not really sure what to do.

Are they executable on your docker image or on the host?
This might be the root of the issue.

**Comment by miguelriemoliveira** [comment]
docker container:

```terminal
drwxrwxr-x 3 mike mike 4.0K Dec 23 15:01 .
drwxrwxr-x 7 mike mike 4.0K Mar 29  2023 ..
-rw-rw-r-- 1 mike mike  982 Nov 20 23:26 configurable_tf_listener.py
-rw-r--r-- 1 mike mike  31K Dec 24 12:29 data_collector.py
-rwxrwxrwx 1 mike mike 8.8K Dec 22 20:30 depth_labeler
-rw-rw-r-- 1 mike mike    0 Mar 28  2023 __init__.py
-rw-r--r-- 1 mike mike  44K Dec 22 20:30 label_messages.py
-rwxrwxrwx 1 mike mike  12K Dec 22 20:30 lidar3d_labeler
-rw-r--r-- 1 mike mike  23K Dec 23 09:24 patterns.py
drwxrwxr-x 2 mike mike 4.0K Dec 24 12:29 __pycache__
-rwxrwxrwx 1 mike mike 8.5K Dec 22 20:30 rgb_labeler
```


Host system:

```terminal
-rw-rw-r-- 1 mike mike  982 Nov 20 23:26 configurable_tf_listener.py
-rw-r--r-- 1 mike mike  31K Dec 24 12:29 data_collector.py
-rwxrwxrwx 1 mike mike 8.8K Dec 22 20:30 depth_labeler
-rw-rw-r-- 1 mike mike    0 Mar 28  2023 __init__.py
-rw-r--r-- 1 mike mike  44K Dec 22 20:30 label_messages.py
-rwxrwxrwx 1 mike mike  12K Dec 22 20:30 lidar3d_labeler
-rw-r--r-- 1 mike mike  23K Dec 23 09:24 patterns.py
drwxrwxr-x 2 mike mike 4.0K Dec 24 12:29 __pycache__
-rwxrwxrwx 1 mike mike 8.5K Dec 22 20:30 rgb_labeler
```

The same in both ... I think distrobox creates a shared filesystem, not sure ...

**Comment by manuelgitgomes** [comment]
I might have found something.

[This stack overflow post](https://stackoverflow.com/questions/59160266/git-and-permissions) explains this very well.
What I gathered from it is that git does not record the permissions of the file.
It records one of two indexes for the file: 755 or 644.
I believe these files were 644, so they are not executable by default.
I changed their index to 755 with the command:
```
git update-index --chmod=+x depth_labeler lidar3d_labeler rgb_labeler
```

Hopefully this works for everyone. 
If anyone finds more cases like this, refer them here so we can see how to solve that.
I will push my changes.
